### ISSUE TITLE
IllegalArgumentException: Class java.lang.Integer not a super-type of [simple type, class int]

### ISSUE THREAD

**Issue by yevsh** [FasterXML/jackson-module-jaxb-annotations#64]
Getting this error when trying to use this module together with jackson-module-jsonSchema module.
My code
```
ObjectMapper mapper = new ObjectMapper();		
JaxbAnnotationModule module = new JaxbAnnotationModule();		
 mapper.registerModule(module);
			 			 
JsonSchemaGenerator schemaGen = new JsonSchemaGenerator(mapper);
JsonSchema schema = schemaGen.generateSchema(IdInfo.class);
```

Full stack:

```
Exception in thread "main" com.fasterxml.jackson.databind.JsonMappingException: Failed to widen type [simple type, class int] with annotation (value java.lang.Integer), from 'getNo': Class java.lang.Integer not a super-type of [simple type, class int]
	at com.fasterxml.jackson.databind.AnnotationIntrospector.refineSerializationType(AnnotationIntrospector.java:821)
	at com.fasterxml.jackson.databind.introspect.AnnotationIntrospectorPair.refineSerializationType(AnnotationIntrospectorPair.java:488)
	at com.fasterxml.jackson.databind.ser.PropertyBuilder.findSerializationType(PropertyBuilder.java:194)
	at com.fasterxml.jackson.databind.ser.PropertyBuilder.buildWriter(PropertyBuilder.java:73)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory._constructWriter(BeanSerializerFactory.java:807)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.findBeanProperties(BeanSerializerFactory.java:608)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.constructBeanSerializer(BeanSerializerFactory.java:388)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.findBeanSerializer(BeanSerializerFactory.java:271)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory._createSerializer2(BeanSerializerFactory.java:223)
	at com.fasterxml.jackson.databind.ser.BeanSerializerFactory.createSerializer(BeanSerializerFactory.java:157)
	at com.fasterxml.jackson.databind.SerializerProvider._createUntypedSerializer(SerializerProvider.java:1215)
	at com.fasterxml.jackson.databind.SerializerProvider._createAndCacheUntypedSerializer(SerializerProvider.java:1187)
	at com.fasterxml.jackson.databind.SerializerProvider.findValueSerializer(SerializerProvider.java:526)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.acceptJsonFormatVisitor(DefaultSerializerProvider.java:409)
	at com.fasterxml.jackson.databind.ObjectWriter.acceptJsonFormatVisitor(ObjectWriter.java:1059)
	at com.fasterxml.jackson.databind.ObjectWriter.acceptJsonFormatVisitor(ObjectWriter.java:1066)
	at com.fasterxml.jackson.module.jsonSchema.JsonSchemaGenerator.generateSchema(JsonSchemaGenerator.java:89)
	at tes.main(tes.java:34)
Caused by: java.lang.IllegalArgumentException: Class java.lang.Integer not a super-type of [simple type, class int]
	at com.fasterxml.jackson.databind.type.TypeFactory.constructGeneralizedType(TypeFactory.java:469)
	at com.fasterxml.jackson.databind.AnnotationIntrospector.refineSerializationType(AnnotationIntrospector.java:819)
```

'getNo' is: 

```
  @XmlElement(type=Integer.class, nillable=true)
  public int getNo()
  {
    return this.no;
  }

```

**Comment by cowtowncoder** [comment]
Exception seems legit: `Integer` and `int` are indeed not compatible types.
Why is `@XmlElement` annotation used here?

**Comment by yevsh** [comment]
Legacy code, generated...

**Comment by cowtowncoder** [comment]
@yevsh Figures. It is possible JAXB accepts that, and there is of course some possible justification since it is wrapper/primitive distinction... and at reflection level wrappers are always used.

I'll have to think about this; I think it should be possible to work out something. But that may take a while.

Now: on short term what you may need to do is to use mix-in annotations to "mask" this annotation.
Or sub-class `JaxbAnnotationIntrospector` and override handling; I think in this case annotation should probably be just ignored in practice.

**Comment by cowtowncoder** [comment]
I filed new issue for `jackson-databind` since I think this is where it would ideally be handled I think.

**Comment by yevsh** [comment]
>  on short term what you may need to do is to use mix-in annotations to "mask" this annotation.
Or sub-class JaxbAnnotationIntrospector and override handling; I think in this case annotation should probably be just ignored in practice.

Can you please elaborate?

**Comment by cowtowncoder** [comment]
@yevsh You could attach different `@XmlElement` via mix-in annotation functionality, and this would prevent original annotation from being read. This would avoid the exception.

**Comment by yevsh** [comment]
and what is this mix-in annotation functionality?

**Comment by cowtowncoder** [comment]
@yevsh Maybe google for it?

**Comment by cowtowncoder** [comment]
As per

https://github.com/FasterXML/jackson-databind/issues/1592

now fixed for 2.9.0 final (not yet included in `2.9.0.pr3`)
